activerecord/lib/active_record/relation/record_fetch_warning.rb

```diff
@@ -23,23 +23,21 @@ def exec_queries
         end
       end
 
-      ActiveSupport::Notifications.subscribe("sql.active_record") do |*args|
-        payload = args.last
-
-        QueryRegistry.queries << payload[:sql]
+      ActiveSupport::Notifications.subscribe("sql.active_record") do |*, payload|
+        QueryRegistry.queries << payload.fetch(:sql)
```

Comment:
Just tried that benchmark on 2.2.1 and I get:

```
Calculating -------------------------------------
                 #[]   139.259k i/100ms
              #fetch   132.383k i/100ms
-------------------------------------------------
                 #[]      7.907M (± 8.3%) i/s -     39.271M
              #fetch      6.944M (± 7.9%) i/s -     34.552M
```

It seems the discrepancy has decreased to only around 10-15% and `fetch` has become 2.25 times faster.